services/nomad/build/master.cfg

```diff
@@ -0,0 +1,324 @@
+for b in builders:
+    name = b['name'] + '_builder'
+    builder_names.append(name)
+
+c['schedulers'].append(schedulers.SingleBranchScheduler(
+    name="all",
+    change_filter=util.ChangeFilter(branch='master'),
+    treeStableTimer=None,
+    builderNames=builder_names))
+
+c['schedulers'].append(schedulers.ForceScheduler(
+    name="force",
+    builderNames=builder_names))
+
+
+# ###### BUILDERS
+
+distdir = 'void-packages'
+hostdir = '/hostdir'
+buildroot = 'buildroot'
+builddir = util.Property('buildnumber')
+
+
+factory = util.BuildFactory()
+
+
+@util.renderer
+def make_xbps_src_cmd(props, cmd):
+    command = [
+        f'{distdir}/xbps-src',
+        '-H', hostdir,
+        '-m', buildroot,
+        '-A', props.getProperty('host'),
+    ]
+
+    if cmd == 'binary-bootstrap':
+        command += shlex.split(str(props.getProperty('bootstrap_args')))
+
+    if props.getProperty('cross') == 'True':
+        command += ['-a', props.getProperty('target')]
+
+    command += [cmd]
+
+    return command
+
+
+@util.renderer
+def make_xbps_src_make_cmd(props):
+    command = [
+        'xbps-src-make',
+        '-hostdir', hostdir,
+        '-distdir', f'../{distdir}',
+        '-masterdir', f'../{buildroot}',
+        '-arch', props.getProperty('host'),
+    ]
+
+    if props.getProperty('cross') == 'True':
+        command += ['-cross', props.getProperty('target')]
+
+    return command
+
+
+class ShellCommandWithChanges(steps.ShellCommand):
+    @defer.inlineCallbacks
+    def run(self):
+        cmd = yield self.makeRemoteShellCommand()
+        cmd.command += self.build.allFiles()
+        yield self.runCommand(cmd)
+        return cmd.results()
+
+
+@util.renderer
+def build_packages(props):
+    # TODO: when a better solver is in place
+    # cmds = []
+    # for p in str(props.getProperty('packages')).strip().split():
+    #     cmds.append(util.ShellArg(
+    #         command=['make', f'built/{p}'],
+    #         logname=f'pkg:{p}',
+    #         haltOnFailure=True,
+    #     ))
+    cmds = [util.ShellArg(
+        command=['make'],
+        logname='build',
+        haltOnFailure=True,
+    )]
+    if cmds:
+        cmds.append(util.ShellArg(
+            command=['make', 'clean'],
+            logname='cleanup',
+            haltOnFailure=True,
+        ))
+    return cmds
+
+
+@util.renderer
+def make_prune_cmd(props):
+    return ['bash', '-c',
+            util.Interpolate(f"""
+export XBPS_TARGET_ARCH="%(prop:target)s"
+for repo in / /debug /nonfree /bootstrap; do
+    xbps-rindex -r "{hostdir}/binpkgs/$repo"
+done
+if [ "$XBPS_TARGET_ARCH" = i686 ]; then
+    for repo in /multilib /multilib/nonfree /multilib/bootstrap; do
+        XBPS_TARGET_ARCH=x86_64 xbps-rindex -r "{hostdir}/binpkgs/$repo"
+    done
+fi
+""")]
+
+
+factory.addStep(steps.Git(
+    # TODO
+    # repourl='https://github.com/void-linux/void-packages.git',
+    repourl='https://github.com/classabbyamp/void-packages.git',
+    mode='incremental',
```

Comment:
What does `incremental` do?

Reply:
> **mode** (optional, default: `'incremental'`): Specifies whether to clean the build tree or not.
> 
> - **incremental** The source is update, but any built files are left untouched.
> - **full**: The build tree is clean of any built files. The exact method for doing this is controlled by the method argument.

Reply:
`incremental` also matches the current buildbot